packages/components/src/modal/Modal.tsx

```diff
@@ -0,0 +1,183 @@
+import ThemeExport from '../ThemeExport';
+
+interface ModalProps {
+  className?: string;
+  children?: ReactNode;
+  role?: string;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  size?: 'sm' | 'lg' | 'xl' | undefined;
+  onOpened?: () => void;
+  onClosed?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}
+
+const Modal = ({
+  className = 'theme-bg-light',
+  children,
+  role = 'role',
+  keyboard = true,
+  isOpen = false,
+  centered = false,
+  size,
+  onOpened,
+  onClosed,
+  toggle,
+  'data-testid': dataTestId,
+}: ModalProps): ReactElement => {
+  const [isPortalOpen, setIsPortalOpen] = useState(false);
+
+  const element = useRef<HTMLElement>();
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case 'Escape':
+          if (toggle) {
+            toggle();
+          }
+          break;
+        default:
+          break;
+      }
+    },
+    [toggle]
+  );
+
+  useEffect(
+    function addKeydownEventListener() {
+      if (isOpen && keyboard) {
+        window.addEventListener('keydown', handleKeyDown);
+        return () => window.removeEventListener('keydown', handleKeyDown);
+      }
+    },
+    [handleKeyDown, isOpen, keyboard]
+  );
+
+  useEffect(
+    function open() {
+      if (isOpen && onOpened) {
+        onOpened();
+      }
+    },
+    [onOpened, isOpen]
+  );
+
+  useEffect(
+    function closed() {
+      if (!isOpen && onClosed) {
+        onClosed();
+      }
+    },
+    [onClosed, isOpen]
+  );
+
+  useEffect(
+    function close() {
+      if (!isPortalOpen) {
+        if (element.current) {
+          document.body.removeChild(element.current);
+          element.current = undefined;
+        }
+      }
+    },
+    [isPortalOpen]
+  );
+
+  useEffect(
+    function open() {
+      if (isOpen && !element.current) {
+        element.current = document.createElement('div');
+        element.current.setAttribute(
+          'style',
+          'z-index: 1050; padding-right: 15px; display: block'
+        );
+        element.current.setAttribute('role', 'modal-container');
+        document.body.appendChild(element.current);
+        setIsPortalOpen(true);
+      }
+    },
+    [isOpen]
+  );
+
+  const onExited = () => {
+    setIsPortalOpen(false);
+  };
+
+  return element.current ? (
+    ReactDOM.createPortal(
+      <>
+        <CSSTransition
+          appear
+          mountOnEnter
+          unmountOnExit
+          in={isOpen}
+          classNames={{
+            enterActive: 'show',
+            enterDone: 'show',
+          }}
+          timeout={ThemeExport.transitionMs}
+          onExited={onExited}
+        >
+          <div className={classNames('modal-backdrop fade')} />
+        </CSSTransition>
+        <CSSTransition
+          appear
+          mountOnEnter
+          unmountOnExit
+          in={isOpen}
+          classNames={{
+            enterActive: 'show',
```

Comment:
I think for content, you only want to specify `enterDone` - that way the content fades in after the background fade transition.